junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/KotlinAssertionsTests.kt

```diff
@@ -60,6 +60,16 @@ class KotlinAssertionsTests {
         assertThrows<AssertionError>({ "should fail" }) { fail(null as Throwable?) }
     }
 
+    @Test
```

Comment:
A general statement for testing this implementation:

There are no scenario's that cover an actual suspending function. Even though it works with suspending functions, I've verified this myself as well, I'm still worried about future adjustments to this code that might change that behaviour. You would never find out because there are no unit tests that will fail. 

I must admit I do not know how to effectively test this either, too. But I think _something_ should be done to avoid future regression issues.

Reply:
Agreed, the tests I've included here only attempt to "simulate" a `CoroutineContext`, short of actually including Coroutines I don't know of a good solution to this problem. Though since it's would only be for test this could be easier to justify, thoughts @marcphilipp ?

Reply:
`kotlinx.coroutines` can be added in test classpath, so then you'll not need this ContextScope

Reply:
> `kotlinx.coroutines` can be added in test classpath, so then you'll not need this ContextScope

Good idea! Using [kotlinx-coroutines-test](https://github.com/Kotlin/kotlinx.coroutines/tree/master/kotlinx-coroutines-test) should work just fine. Hopefully this is allowed from a dependency pollution standpoint.

Reply:
As long as it's on the test classpath only, that's fine!